ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -928,22 +928,43 @@ baseBounds ghcFlavor =
 -- | Common build dependencies.
 commonBuildDepends :: GhcFlavor -> [String]
 commonBuildDepends ghcFlavor =
-  [ "ghc-prim > 0.2 && < 0.8"
-  , baseBounds ghcFlavor
-  , "containers >= 0.5 && < 0.7"
-  , "bytestring >= 0.9 && < 0.11"
-  , "binary == 0.8.*"
-  , "filepath >= 1 && < 1.5"
-  , "directory >= 1 && < 1.4"
-  , "array >= 0.1 && < 0.6"
-  , "deepseq >= 1.4 && < 1.5"
-  , "pretty == 1.1.*"
-  , "time >= 1.4 && < 1.10"
-  , "transformers == 0.5.*"
-  , "process >= 1 && < 1.7"
-  ] ++
-  [ "exceptions == 0.10.*" | ghcFlavor >= Ghc901 ] ++
-  [ "parsec" | ghcFlavor >= Ghc921 ]
+  case ghcFlavor of
+    Ghc921 ->
+      [
+        "ghc-prim > 0.2 && < 0.9"
+      , baseBounds ghcFlavor
+      , "containers >= 0.5 && < 0.7"
+      , "bytestring >= 0.9 && < 0.12"
+      , "binary == 0.8.*"
+      , "filepath >= 1 && < 1.5"
```

Comment:
maybe worth factoring this such that the bounds that are identical are shared?

Reply:
in for a penny, in for a pound i guess 😉